client/src/Todo.js

```diff
@@ -1,25 +1,74 @@
-import React from "react";
+import React, { useState } from "react";
 import { useTodo } from "./TodoProvider";
 import IconButton from "@material-ui/core/IconButton";
+import EditIcon from "@material-ui/icons/Edit";
 import DeleteIcon from "@material-ui/icons/Delete";
+import SaveIcon from "@material-ui/icons/Save";
+import "./styles.css";
 
-export default function Todo({ id, description, complete }) {
-  const { setStatusTodo, deleteTodo } = useTodo();
+export default function Todo({ id, description, completed }) {
+  const { setStatusTodo, deleteSingleTodo, updateTodoDescription } = useTodo();
 
   const checkTodo = (e) => setStatusTodo(id, e.target.checked);
 
+  const [editTodoId, setEditTodoId] = useState(null);
```

Comment:
This component is rendering only one TODO, so there's no need to set an "editId". You can use the ID coming from the props

Reply:
its not working anymore

Reply:
An example would be:
```js
const [isEditMode, setEditMode] = useState(false);

// ...
<button onClick={() => setEditMode(true)} ...>Start edit

const saveEdit = () => {
  // ... do the change where it needs to 

  setEditMode(false);
}

// ...
<button onClick={saveEdit} ...>Save
```